### PR TITLE
add AutoSurrogateLikelihoodDAIS

### PR DESCRIPTION
**Pull request by martinjankowiak**

### CODE REVIEW

**Comment by martinjankowiak** [comment]
@fehiepsi do we need any additional tests or asserts in the code? e.g. making sure the surrogate model has all the (global) latent variables and doesn't have any local latent variables?

File: numpyro/infer/autoguide.py [hunk 1]
```diff
@@ -865,6 +866,208 @@ def _single_sample(_rng_key):
...
+                init_strategy=self.init_loc_fn,
+                dynamic_args=False,
+                model_args=(),
+                model_kwargs={},
+            )
+
+    def _sample_latent(self, *args, **kwargs):
+        def blocked_surrogate_model(x):
+            x_unpack = self._unpack_latent(x)
+            with numpyro.handlers.block(hide_fn=lambda site: site["type"] != "param"):
```
Comment (fehiepsi) [inline numpyro/infer/autoguide.py:988]: Seems like a usage case of `expose_types` in the last PR. :D

File: test/infer/test_autoguide.py [hunk 2]
```diff
@@ -859,3 +860,73 @@ def model2():
...
+        theta = numpyro.sample(
+            "theta", dist.Normal(jnp.zeros(D), jnp.ones(D)).to_event(1)
+        )
+        omegas = numpyro.param(
+            "omegas", 2.0 * jnp.ones(N // 2), constraint=dist.constraints.positive
+        )
+
+        with numpyro.plate("N", N // 2), numpyro.handlers.scale(scale=omegas):
+            X_batch = numpyro.subsample(X, event_dim=1)
+            Y_batch = numpyro.subsample(Y, event_dim=0)
```
Comment (fehiepsi) [inline test/infer/test_autoguide.py:885]: I think we can use `X`, `Y` here for clarity or change the signature of this model to `X_batch`, `Y_batch` and inside the model, replace `N // 2` by `M = Y_batch.shape[0]`.

**Review by fehiepsi — DISMISSED** [review]
LGTM! Just have minor comments

**Review by fehiepsi — APPROVED** [review]
LGTM!

**Pull request merged**